pilot/pkg/model/push_context.go

```diff
@@ -243,6 +243,9 @@ func (ps *PushContext) UpdateMetrics() {
 // VirtualServices lists all virtual services bound to the specified gateways
 // This replaces store.VirtualServices
 func (ps *PushContext) VirtualServices(gateways map[string]bool) []Config {
+	ps.mutex.Lock()
+	defer ps.mutex.Unlock()
+
```

Comment:
this seems pointless. PushContext is per sidecar. Why even lock/unlock this? If anything, it will add overhead won't it?

Reply:
@rshriram This is not the cause and will be reverted once I know how to fix the issue.
Basically the PushContext's initVirtualServices() is modifying the VS Config with FQDN while Monitor is reading it for old/new comparison.
Any suggestion how to sync the two?
I thought about making a copy of the vservices in PushContext but I'm not sure it's the right approach.